boundary/analyze.py

```diff
@@ -194,26 +193,53 @@ def check_redirect_mangled(f, meta):
         if to_sym != f or to_sym[1] != from_sym[1]:
             continue
         # Unsafe if the caller is a sched_outsider
-        if from_sym in func_class['sched_outsider']:
+        if from_sym in func_class.sched_outsider:
             return True
         # If the caller is optimized too, check if it's unsafe recursively
-        if from_sym in func_class['mangled'] or from_sym not in func_class['in_vmlinux']:
+        if from_sym in func_class.mangled or from_sym not in func_class.in_vmlinux:
             if check_redirect_mangled(from_sym, meta):
                 return True
     return False
 
+def func_class_arithmetics(fns):
+    fns.callback -= fns.interface
```

Comment:
commit log:

border: use dotdict and underscope to shorten lines  --> there is no underscope now. (fns)